daffodil-runtime1/src/main/scala/org/apache/daffodil/dpath/DPath.scala

```diff
@@ -160,6 +161,12 @@ final class RuntimeExpressionDPath[T <: AnyRef](qn: NamedQName, tt: NodeInfo.Kin
             case _ => doSDE(e, ps)
           }
         }
+        case ParserNonBlocking => {
+          e match {
+            case nd: InfosetNoDataException => doSDE(nd, ps)
```

Comment:
This doesn't change anything to use the new exception. I think you want something like
```scala
case nd: InfosetNoDataException => {
  val selfRefException = new InfosetSelfReferencingException(nd.node)
  doSDE(selfRefException, ps)
}
```

Reply:
Right, I implemented your InfosetNoDataExceptionBase suggestion. That arrears to have corrected the IT error and improved test flexibility.